src/java.base/share/classes/java/io/RandomAccessFile.java

```diff
@@ -80,6 +85,36 @@ public class RandomAccessFile implements DataOutput, DataInput, Closeable {
     private static final int O_DSYNC =  8;
     private static final int O_TEMPORARY =  16;
 
+    private class Resource implements JDKResource {
+        Resource() {
+            Core.getJDKContext().register(this);
+        }
+
+        @Override
+        public void beforeCheckpoint(Context<? extends jdk.crac.Resource> context) throws Exception {
+            if (Core.getJDKContext().claimFdWeak(fd, this)) {
+                if (Core.getJDKContext().matchClasspath(path)) {
```

Comment:
Actually it would be more natural to match with PersistentJarFile who claimed their FDs already. So we'll be able to handle dynamically loaded JARs that are not a part of the `java.class.path`

Reply:
This case is trying to prevent a situation when the FD is **not** opened using `PersistentJarFile`, as Spring Boot uses own implementation.

Reply:
Indeed! This is a required transition from the old logic in hotspot, which ignored all files (by path) specified in classpath. 

I was thinking about another use-case, when we have an instance of `PersistentJarFile` (which allows underlying fd), and RandomAccessFile with another underlying fd, but reffering to the same path on file system.